### DISCUSSION TITLE
UMAT as Plugin

### DISCUSSION THREAD

**Discussion by 12arya129** [idaholab/moose#17165] [Q&A Modules: General] [answered]
Hello,

I want to learn more about using UMAT (Fortran) in moose, I tried searching it on Moose website but couldn't find much. Any help?

**Comment by lynnmunday** [comment, marked as answer]
It was a part of the MOOSE solid mechanics module but the solid mechanics module was deprecated in favor of the tensor mechanics module and the umat code did not make it into tensor mechanics.  If you wanted to try reimplementing the umat code into tensor mechanics, you could start from this commit from 7/25/2012 which implemented umat in solid mechanics:  94051c7ae    

I think we might try to add a umat capability back into tensor mechanics this summer.

**Reply by Traiwit** [reply]
Hi @lynnmunday is there any update on UMAT?

**Reply by lynnmunday** [reply]
@dschwen has done a lot of umat work recently.   In May he merged this:  
https://github.com/idaholab/moose/pull/17742
There are also some other issues he is working on.  I think it would be ready to try and there are examples in the above pr.

**Reply by dschwen** [reply]
Stay tuned for further improvements in that area. And let us know if you have specific UMAT requirements. We're always interested in feedback on this capability!
